collect_app/src/main/java/org/odk/collect/android/widgets/warnings/SpacesInUnderlyingValuesWarning.java

```diff
@@ -0,0 +1,142 @@
+
+    public static SpacesInUnderlyingValuesWarning forQuestionWidget(QuestionWidget questionWidget) {
+        WarningRenderer renderer = new RenderIntoQuestionWidget(questionWidget);
+        SpacesInUnderlyingValues valuesChecker = new SpacesInUnderlyingValuesImpl();
+        return new SpacesInUnderlyingValuesWarning(valuesChecker, renderer);
+    }
+
+    @VisibleForTesting
+    SpacesInUnderlyingValuesWarning(SpacesInUnderlyingValues valuesChecker, WarningRenderer warningRenderer) {
+        this.valuesChecker = valuesChecker;
+        this.warningRenderer = warningRenderer;
+    }
+
+
+    public void renderWarningIfNecessary(List<SelectChoice> items) {
+        valuesChecker.check(items);
+
+        if (valuesChecker.hasInvalidValues()) {
+            warningRenderer.render(valuesChecker.getInvalidValues());
+        }
+    }
+
+    interface WarningRenderer {
+        void render(List<SelectChoice> items);
+    }
+
+    private static class RenderIntoQuestionWidget implements WarningRenderer {
+
+        private QuestionWidget questionWidget;
+        private final WarningTextCreator warningCreator;
+
+        RenderIntoQuestionWidget(QuestionWidget questionWidget) {
+            this.questionWidget = questionWidget;
+            warningCreator = new WarningTextCreatorImpl();
+        }
+
+        @Override
+        public void render(List<SelectChoice> invalidItems) {
+            questionWidget.showWarning(warningCreator.create(invalidItems, questionWidget.getContext()));
+        }
+    }
+
+    interface SpacesInUnderlyingValues {
+        void check(List<SelectChoice> items);
+
+        boolean hasInvalidValues();
+
+        List<SelectChoice> getInvalidValues();
+    }
+
+    public static class SpacesInUnderlyingValuesImpl implements SpacesInUnderlyingValues {
+
+        private List<SelectChoice> invalidValues = Lists.newArrayList();
+        private boolean checked = false;
+
+        @Override
+        public void check(List<SelectChoice> items) {
+            invalidValues = FluentIterable
+                    .from(items)
+                    .filter(item -> item.getValue() != null && item.getValue().contains(" "))
+                    .toList();
+            checked = true;
+        }
+
+        @Override
+        public boolean hasInvalidValues() {
+            checkInitialization();
+            return !invalidValues.isEmpty();
+        }
+
+        @Override
+        public List<SelectChoice> getInvalidValues() {
+            checkInitialization();
+            return invalidValues;
+        }
+
+        private void checkInitialization() {
+            if (!checked) {
+                throw new IllegalStateException("check() must be called before other methods first");
+            }
+        }
+    }
+
+    public interface WarningTextCreator {
+        String create(List<SelectChoice> invalidValues, Context context);
+    }
+
+    private static class WarningTextCreatorImpl implements WarningTextCreator {
```

Comment:
Instead of `WarningTextCreatorImpl` how about something like `SpacesWarningTextCreator`?